sbin/pfctl/pfctl.c

```diff
@@ -2770,7 +2770,7 @@ main(int argc, char *argv[])
 	if ((path = calloc(1, MAXPATHLEN)) == NULL)
 		errx(1, "pfctl: calloc");
 	memset(anchorname, 0, sizeof(anchorname));
-	if (anchoropt != NULL) {
+	if (anchoropt != NULL && len > 0) {
 		int len = strlen(anchoropt);
 
 		if (anchoropt[len - 1] == '*') {
```

Comment:
```suggestion
		if (len >= 1 && anchoropt[len - 1] == '*') {
```
preserves current behaviour on non-CHERI (with the exception of if there happens to be a `*` just before the string in memory, but given anchoropt is optarg I suspect it's always the `'\0'` of the preceding argument). Otherwise loadopt will change value.